### DISCUSSION TITLE
Alerts Index?

### DISCUSSION THREAD

**Discussion by dking986** [Security-Onion-Solutions/securityonion#3919] [Unsupported Versions] [answered]
Hello,

Where are alerts stored in SO? I am digging through my indicies looking for documents that correlate to my alerts, but just finding the normal events. Is there a special index or location they are stored, or a special way to query them?

**Comment by weslambert** [comment, marked as answer]
Alerts are stored in various indices, depending on what type of data it is (ids, ossec, etc).  You can query alerts in a free-form fashion just using `event.dataset:alert`.  If you would like to view alerts within a queue, try navigating to the `Alerts` pane within Security Onion Console.

**Reply by dking986** [reply]
OK, I am trying to query alerts in python via elasticsearch api, to feed into various things. Is there an api I can hook into or fields I can pivot off of to aggregate the alerts from the different indices.

**Reply by weslambert** [reply]
Aggregation would need to be performed by your query.  `event.dataset:alert` would span any events across all indices with a dataset value of `alert`

**Comment by dking986** [comment]
Update: 
Got it working, here is the python snippet. This pulls events for the last 24 hours.

```
body = {
    "query": {
        "bool": {
            "must": [
                {
                    "match": {
                        "event.dataset": "alert"
                    }
                }
            ],
            "filter": [
                {
                    "range": {
                        "@timestamp": {
                            "gte": "now-1d",
                            "lte": "now"
                        }
                    }
                }
            ]
        }
    }
}

results = es.search(index="*", timeout="2m", size=10000, body=body)
```
